server/src/main/java/org/elasticsearch/transport/NoSuchRemoteClusterException.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.transport;
+
+/**
+ * An exception that remote cluster is missing or
+ * connectivity to the remote connection is failing
+ */
+public final class NoSuchRemoteClusterException extends IllegalArgumentException {
+
+    NoSuchRemoteClusterException(String clusterName) {
+        //No node available for cluster
+        super("no such remote cluster: [" + clusterName + "]");
```

Comment:
we usually pass the resource to the metadata with `addMetadata` and other exceptions like this subclass `ResourceNotFoundException` WDYT?

Reply:
I did think about this, but because this exception does not get serialized and always handled on the same node, I found that handeling the exception handeling is cleaner with a dedicated exception. (otherwise a `instanceof` check and a metadata check is required). 

I always thought that we preferred adding metadata to exceptions that get serialized. Maybe I'm not recalling that correctly. If you feel strongly about this then I will throw a `ResourceNotFoundException` exception with metadata.

Reply:
maybe just subclass `ResourceNotFoundException` and we are good